src/overlays/actors/ovl_En_Floormas/z_en_floormas.c

```diff
@@ -9,123 +15,1118 @@ void EnFloormas_Destroy(Actor* thisx, GlobalContext* globalCtx);
+    this->actor.shape.rot.x = -0x4CE0;
+    this->actionFunc = EnFloormas_GrabLink;
+}
+
+void EnFloormas_MergeBegin(EnFloormas* this) {
+    SkelAnime_ChangeAnimDefaultStop(&this->skelAnime, &D_06009DB0);
+    this->actionTimer = 0;
+    this->smActionTimer += 1500;
+    EnFloormas_Invulnerable(this);
+    this->actionFunc = EnFloormas_Merge;
+}
+
+void EnFloormas_SmWaitBegin(EnFloormas* this) {
+    EnFloormas* attachedA = (EnFloormas*)this->actor.attachedA;
+    EnFloormas* attachedB = (EnFloormas*)this->actor.attachedB;
+
+    // if this is the last remaining small floor master, kill all.
+    if ((attachedA->actionFunc == EnFloormas_SmWait) && (attachedB->actionFunc == EnFloormas_SmWait)) {
+        Actor_Kill(&attachedA->actor);
+        Actor_Kill(&attachedB->actor);
+        Actor_Kill(&this->actor);
+        return;
+    }
+    this->actor.draw = NULL;
+    this->actionFunc = EnFloormas_SmWait;
+    this->actor.flags &= ~0x11;
+}
+
+void EnFloormas_TakeDamageBegin(EnFloormas* this) {
+    SkelAnime_ChangeAnimTransitionStop(&this->skelAnime, &D_06000590, -3.0f);
+    if ((this->collider.body.acHitItem->toucher.flags & 0x1F824) != 0) {
+        this->actor.posRot.rot.y = this->collider.base.ac->posRot.rot.y;
+    } else {
+        this->actor.posRot.rot.y = func_8002DA78(&this->actor, this->collider.base.ac) + 0x8000;
+    }
+    func_8003426C(&this->actor, 0x4000, 0xFF, 0, 0x14);
+    this->actionFunc = EnFloormas_TakeDamage;
+    this->actor.speedXZ = 5.0f;
+    this->actor.velocity.y = 5.5f;
+}
+
+void EnFloormas_RecoverBegin(EnFloormas* this) {
+    SkelAnime_ChangeAnimDefaultStop(&this->skelAnime, &D_06000EA4);
+    this->actor.velocity.y = this->actor.speedXZ = 0.0f;
+    this->actor.posRot.rot.y = this->actor.shape.rot.y;
+    this->actionFunc = EnFloormas_Recover;
+}
+
+void EnFloormas_FreezeBegin(EnFloormas* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_060019CC, 1.5f, 0, 20.0f, 2, -3.0f);
+    this->actor.speedXZ = 0.0f;
+    if (this->actor.colChkInfo.damageEffect == 4) {
+        func_8003426C(&this->actor, -0x8000, 0xFF, 0, 0x50);
+    } else {
+        func_8003426C(&this->actor, 0, 0xFF, 0, 0x50);
+        if (this->actor.scale.x > 0.004f) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_FREEZE);
+        } else {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_FREEZE);
+        }
+    }
+    this->actionTimer = 80;
+    this->actionFunc = EnFloormas_Freeze;
+}
+
+void EnFloormas_Die(EnFloormas* this, GlobalContext* globalCtx) {
+    // split
+    if (this->actor.scale.x > 0.004f) {
+        this->actor.shape.rot.y = this->actor.rotTowardsLinkY + 0x8000;
+        EnFloormas_SplitBegin((EnFloormas*)this->actor.attachedB);
+        EnFloormas_SplitBegin((EnFloormas*)this->actor.attachedA);
+        EnFloormas_SplitBegin(this);
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_FLOORMASTER_SPLIT);
+        // Die
```

Comment:
I suggest moving this kind of comment (like the ones above and below) inside the actual blocks to avoid the weird alignment.